app/src/organisms/Devices/RobotSettings/AdvancedTab/RobotServerVersion.tsx

```diff
@@ -35,14 +37,35 @@ export function RobotServerVersion({
   const { autoUpdateAction } = useSelector((state: State) => {
     return getBuildrootUpdateDisplayInfo(state, robotName)
   })
+  const [
+    showSoftwareUpdateModal,
+    setShowSoftwareUpdateModal,
+  ] = React.useState<boolean>(false)
+
+  const handleLaunchRobotUpdateModal = (): void => {
+    setShowSoftwareUpdateModal(true)
+  }
   const robotServerVersion =
     robot?.status != null ? getRobotApiVersion(robot) : null
 
   return (
     <>
+      {showSoftwareUpdateModal &&
+      robot != null &&
+      robot.status !== UNREACHABLE ? (
+        <Portal level="top">
+          <UpdateBuildroot
+            robot={robot}
+            close={() => setShowSoftwareUpdateModal(false)}
+          />
+        </Portal>
+      ) : null}
       {autoUpdateAction !== 'reinstall' && robot != null ? (
```

Comment:
Currently RobotSettings checks two thing (useCurrentRunId() and useAllSessionsQuery() ) to prevent unexpected issue. This component also needs to check them. 
When the robot isn't used by anyone or by any actions, we can show the banner. 
**Already talked about this with Brian**

```ts
import { useRobot, useIsRobotBusy } from '../../hooks'
const isBusy = useIsRobotBusy()
```

```suggestion
      {autoUpdateAction !== 'reinstall' && robot != null && !isBusy ? (
```

Reply:
Thanks for letting me know! Adding it now